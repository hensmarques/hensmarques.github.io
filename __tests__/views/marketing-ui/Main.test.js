import React from 'react';
import { Main } from '../../../src/views/marketing-ui/Main';

describe('<Main/>', () => {
  let wrapper;
  const mockProps = {
    setDisplayMobile: jest.fn()
  }
  beforeEach(() => {
    wrapper = shallow(<Main location={{ modal: null }} {...mockProps} />);
  });

  it('renders and matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('blacklist mode', () => {
    it('renders the BlacklistModal', () => {
      const wrapper = shallow(
        <Main location={{ modal: null }} blacklistMode {...mockProps} />
      );
      expect(wrapper.find('Connect(Nav)').props().blacklistMode).toBe(true);
      expect(wrapper.find('BlacklistModal').exists()).toBe(true);
      expect(wrapper.find('Connect(SignupModal)').exists()).toBe(false);
      expect(wrapper.find('Connect(LoginModal)').exists()).toBe(false);
    });
  });
});
