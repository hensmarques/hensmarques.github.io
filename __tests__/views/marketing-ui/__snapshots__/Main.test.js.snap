// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> renders and matches the snapshot 1`] = `
<div
  id="top"
>
  <div
    id="splash-hero-banner-wrapper"
  >
    <Connect(Nav) />
    <div
      className="top-container"
    >
      <img
        className="main-logo"
        src="images/logos/company-logo-white.png"
      />
    </div>
    <div
      className="mobileLogin only-show-on-mobile"
    >
      <Link
        replace={false}
        to="/login"
      >
        <button
          className="login-btn"
          to="/login"
        >
          Login
        </button>
      </Link>
      <Link
        className="signup"
        replace={false}
        to="/signup"
      >
        Sign Up
      </Link>
    </div>
  </div>
  <Features />
  <BuyCrypto />
  <About />
  <Footer />
</div>
`;
